### DISCUSSION TITLE
Resolver return value type is not fully checked against the `output` schema

### DISCUSSION THREAD

**Discussion by dodas** [trpc/trpc#1817] [Q&A] [answered]
Hello and thanks for the great work on trpc!

I am trying to use the new `output` option for query and expecting to get a TS error when the `resolve` function doesn't return matching type.

Example:
```ts
const router = trpc.router();

router.query("test", {
  output: z.object({
    propA: z.string(),
  }),
  resolve() {
    // missing `propA`
    return {};
  },
});
```

I expect TS to say `resolve()`'s return type is missing `propA`, but it's not happening, it compiles just fine.
Returning for example `true` in `resolve()` does cause a TS error, but returning an object with missing properties does not.

**Comment by jlalmes** [comment]
Hi @dodas, can you please confirm that you have set `strict: true` in your `tsconfig.json`?

**Comment by dodas** [comment, marked as answer]
Yes, I do.
Please see the reproduction here: https://stackblitz.com/edit/typescript-zqmach?file=index.ts

<img width="980" alt="Screenshot 2022-04-22 at 19 12 49" src="https://user-images.githubusercontent.com/14360171/164762639-9c6bfb96-2c93-4274-815d-d5f88079b75d.png">

**Reply by jlalmes** [reply]
Thanks for the reproduction, I will investigate at some point over the next few days and get back to you!

**Reply by KATT** [reply]
THANKS @jlalmes for fixing this! Let us know if the next minor works, @dodas

https://github.com/trpc/trpc/releases/tag/v9.23.0

**Reply by dodas** [reply]
Oh, what an interesting TS quirk, TIL.

I can confirm the issue is fixed in 9.23.0.
Thanks!
